test/NuGet.Core.FuncTests/NuGet.Protocol.FuncTest/AuthenticationHandlerTests.cs

```diff
@@ -0,0 +1,212 @@
+            }
+
+            public IReadOnlyList<HttpListenerRequest> Requests
+            {
+                get
+                {
+                    return _requests;
+                }
+            }
+
+            public void Start(int port)
+            {
+                if (_httpListener != null)
+                {
+                    throw new InvalidOperationException("Already started");
+                }
+
+                string baseUrl = $"http://localhost:{port}/";
+                var listener = new HttpListener();
+                listener.Prefixes.Add(baseUrl);
+                listener.AuthenticationSchemes = AuthenticationSchemes.Basic | AuthenticationSchemes.Anonymous;
+                listener.Start();
+                _httpListener = listener;
+                _baseUrl = baseUrl;
+
+                listener.BeginGetContext(EndGetContext, listener);
+            }
+
+            public void EndGetContext(IAsyncResult result)
+            {
+                HttpListener httpListener = (HttpListener)result.AsyncState!;
+                if (httpListener.IsListening)
+                {
+                    HttpListenerContext? context = null;
+                    try
+                    {
+                        context = httpListener.EndGetContext(result);
+
+                        _requests.Add(context.Request);
+
+                        string? authorization = context.Request.Headers["Authorization"];
+
+                        if (authorization == null)
+                        {
+                            context.Response.StatusCode = 401;
+                            context.Response.AddHeader("WWW-Authenticate", "Basic");
+                        }
+                        else
+                        {
+                            context.Response.StatusCode = 200;
+                        }
+
+                        _output.WriteLine($"Got request for {context.Request.Url}. Auth: {authorization}");
+                    }
+                    catch (Exception ex)
+                    {
+                        if (context != null)
+                        {
+                            context.Response.StatusCode = 500;
+                            using (var textStream = new StreamWriter(context.Response.OutputStream, Encoding.UTF8))
+                            {
+                                textStream.Write(ex.ToString());
+                            }
+                        }
+                    }
+
+                    context?.Response.Close();
+
+                    httpListener.BeginGetContext(EndGetContext, httpListener);
+                }
+            }
+
+            public void Stop()
+            {
+                if (_httpListener == null)
+                {
+                    throw new InvalidOperationException("Not started");
+                }
+
+                var listener = _httpListener;
+                _httpListener = null;
+                _baseUrl = null;
```

Comment:
I see a test added in the PR failed `Cannot access a disposed object.` Is it because `_baseURL` is set to null before invoking `Stop` and `Close` on the listener?

Reply:
no, it's just differences between how HttpListener works on .NET(Core) vs .NET Framework 😞